src/gtest/test_sideChain.cpp

```diff
@@ -0,0 +1,1196 @@
+
+    //checks
+    EXPECT_TRUE(res);
+}
+
+TEST_F(SideChainTestSuite, DuplicatedScCreationsAreNotApplicableToState) {
+    //insert a sidechain
+    uint256 newScId = uint256S("1492");
+    CAmount initialFwdAmount = 1953;
+    aTransaction = createSideChainTxWith(newScId, initialFwdAmount);
+    coinViewCache.UpdateScInfo(aTransaction, aBlock, anHeight);
+
+    CAmount anotherFwdTransfer = 1815;
+    CTransaction duplicatedTx = createSideChainTxWith(newScId, anotherFwdTransfer);
+
+    //Prerequisite
+    ASSERT_TRUE(coinViewCache.sidechainExists(newScId))
+        <<"Test requires the Sc creation tx to be new";
+
+    //test
+    bool res = sideChainManager.IsTxApplicableToState(duplicatedTx, &coinViewCache);
+
+    //checks
+    EXPECT_FALSE(res);
+}
+
+TEST_F(SideChainTestSuite, ForwardTransfersToExistingSCsAreApplicableToState) {
+    //insert a sidechain
+    uint256 newScId = uint256S("1492");
+    CAmount initialFwdAmount = 1953;
+    aTransaction = createSideChainTxWith(newScId, initialFwdAmount);
+    coinViewCache.UpdateScInfo(aTransaction, aBlock, anHeight);
+
+    CAmount aFwdTransfer = 5;
+    aTransaction = createFwdTransferTxWith(newScId, aFwdTransfer);
+
+    //Prerequisite
+    ASSERT_TRUE(coinViewCache.sidechainExists(newScId))
+        <<"Test requires the Sc creation tx to be new";
+
+    //test
+    bool res = sideChainManager.IsTxApplicableToState(aTransaction, &coinViewCache);
+
+    //checks
+    EXPECT_TRUE(res);
+}
+
+TEST_F(SideChainTestSuite, ForwardTransfersToNonExistingSCsAreNotApplicableToState) {
+    uint256 nonExistentScId = uint256S("1492");
+
+    CAmount aFwdTransfer = 1815;
+    aTransaction = createFwdTransferTxWith(nonExistentScId, aFwdTransfer);
+
+    //Prerequisite
+    ASSERT_FALSE(coinViewCache.sidechainExists(nonExistentScId))
+        <<"Test requires target sidechain to be non-existent";
+
+    //test
+    bool res = sideChainManager.IsTxApplicableToState(aTransaction, &coinViewCache);
+
+    //checks
+    EXPECT_FALSE(res);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+//////////////////////////// IsTxAllowedInMempool /////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+TEST_F(SideChainTestSuite, EmptyTxsAreAllowedInEmptyMemPool) {
+    aTransaction = createEmptyScTx();
+
+    //prerequisites
+    ASSERT_TRUE(aMemPool.size() == 0)<<"Test requires empty mempool";
+    ASSERT_TRUE(aTransaction.ccIsNull())<<"Test requires not Sc creation tx, nor forward transfer tx";
+    ASSERT_TRUE(txState.IsValid())<<"Test require transition state to be valid a-priori";
+
+    //test
+    bool res = sideChainManager.IsTxAllowedInMempool(aMemPool, aTransaction, txState);
+
+    //check
+    EXPECT_TRUE(res);
+    EXPECT_TRUE(txState.IsValid());
+}
+
+TEST_F(SideChainTestSuite, EmptyTxsAreAllowedInNonEmptyMemPool) {
+    aTransaction = createEmptyScTx();
+
+    CAmount txFee;
+    double txPriority;
```

Comment:
txPriority is formally an input, it should be initialized

Reply:
Agree. Initialized to 0.0 as not really used in test. Following txPriority too